From 4e7c3c5f8e47d9acc4df689819c57e2e7968e6fe Mon Sep 17 00:00:00 2001
From: Tiago Vignatti <tiago.vignatti@intel.com>
Date: Mon, 15 Jul 2013 17:00:13 -0300
Subject: [PATCH 2/8] ui/aura: Initialize RootWindowHostOzone with native
 AcceleratedWidget

It also make sure the size of the window is corresponding with the native
widget implementation (AttemptToResizeAcceleratedWidget).
---
 ui/aura/root_window_host_ozone.cc |   11 ++++++++---
 ui/aura/root_window_host_ozone.h  |    1 +
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/ui/aura/root_window_host_ozone.cc b/ui/aura/root_window_host_ozone.cc
index 2e046bc..84a484c 100644
--- a/ui/aura/root_window_host_ozone.cc
+++ b/ui/aura/root_window_host_ozone.cc
@@ -5,12 +5,19 @@
 #include "ui/aura/root_window_host_ozone.h"
 
 #include "ui/aura/root_window.h"
+#include "ui/base/ozone/surface_factory_ozone.h"
 
 namespace aura {
 
 RootWindowHostOzone::RootWindowHostOzone(const gfx::Rect& bounds)
     : delegate_(NULL), bounds_(bounds), factory_(new ui::EventFactoryOzone()) {
   factory_->CreateEvdevWatchers();
+  ui::SurfaceFactoryOzone* surface_factory =
+    ui::SurfaceFactoryOzone::GetInstance();
+  widget_ = (gfx::AcceleratedWidget) surface_factory->GetAcceleratedWidget();
+
+  surface_factory->AttemptToResizeAcceleratedWidget(widget_, bounds_);
+
   base::MessagePumpOzone::Current()->AddDispatcherForRootWindow(this);
 }
 
@@ -39,9 +46,7 @@ RootWindow* RootWindowHostOzone::GetRootWindow() {
 }
 
 gfx::AcceleratedWidget RootWindowHostOzone::GetAcceleratedWidget() {
-  // TODO(rjkroege): Support more than one screen. Screen ids start at 1 and
-  // increase for each screen.
-  return (gfx::AcceleratedWidget)1;
+  return widget_;
 }
 
 void RootWindowHostOzone::Show() { NOTIMPLEMENTED(); }
diff --git a/ui/aura/root_window_host_ozone.h b/ui/aura/root_window_host_ozone.h
index 107f2bf..f9c6ad7 100644
--- a/ui/aura/root_window_host_ozone.h
+++ b/ui/aura/root_window_host_ozone.h
@@ -54,6 +54,7 @@ class RootWindowHostOzone : public RootWindowHost,
   virtual void PrepareForShutdown() OVERRIDE;
 
   RootWindowHostDelegate* delegate_;
+  gfx::AcceleratedWidget widget_;
   gfx::Rect bounds_;
 
   // EventFactoryEvdev reads from /dev/input/* file descriptors and creates
-- 
1.7.9.5

